### PR TITLE
Configurable bindings

### PR DESCRIPTION
**Pull request by timvlaer**
With this extension, the plugin user can add JAXB bindings to the XJC process. 
My fork (and so this pull request apparently) is based on the latest version of @theBlackDragon with the most recent jaxb libraries.

### CODE REVIEW

**Comment by theBlackDragon** [comment]
I'm a bit late to the party in replying to this but using something like the following you could already get the plugin to pick up all binding files in a directory:

```scala
xjcCommandLine ++= Seq("-extension", "-b", "src/main/resources/xml-schemas")
```
This is not exactly well documented though (the xjc manpage claims you need one -b per binding, though most implementations seem to conveniently support just picking them all up with a non-standard extension, hence the "-extension" argument).

Note that the "old" libraries used by current master don't support this, hence why I updated the versions in my fork (I probably should've documented that in the commit messages, mea culpa).

**Comment by benmccann** [comment]
+1 great change!  @retronym could we commit this?

File: src/main/scala/com/github/retronym/sbtxjc/SbtXjcPlugin.scala [hunk 1]
```diff
@@ -19,16 +19,18 @@ object SbtXjcPlugin extends Plugin {
...
+  val xjcSettings     = Seq[Def.Setting[_]](
     ivyConfigurations ++= Seq(XjcTool, XjcPlugin),
     xjcCommandLine    := Seq(),
+    xjcBindings       := Seq(),
     xjcPlugins        := Seq(),
     xjcLibs           := Seq(
-      "javax.xml.bind" % "jaxb-api" % "2.1",
-      "com.sun.xml.bind" % "jaxb-impl" % "2.1.9",
-      "com.sun.xml.bind" % "jaxb-xjc" % "2.1.9"
+      "javax.xml.bind" % "jaxb-api" % "2.2.7",
```
Comment (benmccann) [inline src/main/scala/com/github/retronym/sbtxjc/SbtXjcPlugin.scala]: Perhaps we should use 2.2.11 now?

File: project/plugins.sbt [hunk 2]
```diff
@@ -5,3 +5,5 @@ resolvers += Resolver.url("Typesafe snapshot repository", new java.net.URL("http
 libraryDependencies <+= (sbtVersion).apply { sv =>
```
Comment (benmccann) [inline project/plugins.sbt]: The couple lines before this look like a really old syntax and URL. How about:

    resolvers += "Typesafe repository" at "https://repo.typesafe.com/typesafe/releases/"

**Comment by retronym** [comment]
@timvlaer / @theBlackDragon Thanks for contributing! Sorry I didn't see the pull request until now.

@benmccann Thanks for reviewing! I wonder if you'd be interested taking over this project? I've invited you to a team with merge permissions.

**Comment by benmccann** [comment]
Thanks @retronym. I believe I'd accepted the invite, but it looks like I don't have access to merge this yet. Perhaps that team doesn't actually have merge permissions on this project?

**Comment by retronym** [comment]
I'd created the team but forgotten to add it to the repo. Just did so now, try again!

**Comment by benmccann** [comment]
Cool. Thanks!

File: src/main/scala/com/github/retronym/sbtxjc/SbtXjcPlugin.scala (hunk 1 above)
Comment (benmccann) [inline src/main/scala/com/github/retronym/sbtxjc/SbtXjcPlugin.scala]: I tested 2.2.11 and found they screwed up their pom file, so maybe it'd be easier to wait to upgrade until they fix that. I filed a bug for it here: https://java.net/jira/browse/JAXB-1070

**Comment by benmccann** [comment]
I just checked in a test for this code:
https://github.com/sbt/sbt-xjc/commit/41dc2f9e4daec7e452420dc589c0833ca530b943

**Pull request merged**